scripts/other-yes-list/parse.ts

```diff
@@ -0,0 +1,63 @@
+export type Course = {
+  /** Corresponds to the "prefix" column on the PDF */
+  subject: string;
+  /** Corresponds to the "code" column on the PDF */
+  catalogNbr: number;
+  /** The liberal studies categories satisfied by the course */
+  categories: readonly string[];
+};
+
+// creates list of course entries
+export function genCourseEntries(data) {
+  const courseEntries: Course[] = [];
+  let col = 'course prefix';
+  let subject = '';
+  let catalogNbr = 0;
+  let categories = [];
+
+  // find the index of the first course (right after Notes)
+  let start = 0;
+  for (let i = 0; i < data.length; i += 1) {
+    try {
+      const v = data[i][0].str.trim();
+      if (v.slice(0, 5) === 'Notes') {
+        start = i + 1;
+      }
+    } catch {
+      // in case there is no list of approved courses
+    }
+  }
+  // assumes that the columns are ordered and there are no empty cells
+  for (let i = start; i < data.length; i += 1) {
+    try {
+      const v = data[i][0].str.trim();
+      if (col === 'course prefix' && v.length <= 5) {
```

Comment:
What is the `<= 5` check here for?

Reply:
It's the logic in order to know when there is a new row because only the prefix column will always less than 5 characters.